### PR TITLE
[stdlib] Replace `Pointer` by `UnsafePointer` in `hash|simd.mojo`

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**

### CODE REVIEW

File: stdlib/src/builtin/hash.mojo [hunk 1]
```diff
@@ -63,14 +63,16 @@ fn _HASH_SECRET() -> Int:
...
 
-fn _initialize_hash_secret(payload: Pointer[NoneType]) -> Pointer[NoneType]:
+fn _initialize_hash_secret(
+    payload: UnsafePointer[NoneType],
+) -> UnsafePointer[NoneType]:
     var secret = random.random_ui64(0, UInt64.MAX)
-    var data = Pointer[Int].alloc(1)
-    data.store(int(secret))
+    var data = UnsafePointer[Int].alloc(1)
+    initialize_pointee_move(data, int(secret))
```
Comment (soraros) [inline stdlib/src/builtin/hash.mojo:71]: I think we don't need this for reg types.

Comment (gabrieldemarmiesse) [reply stdlib/src/builtin/hash.mojo:71]: What code change would you recommend?

Comment (gabrieldemarmiesse) [reply stdlib/src/builtin/hash.mojo]: Done!

**Review by JoeLoser — APPROVED** [review]
Looks good, thank you for continuing to push on this!

**Comment by JoeLoser** [comment]
Closing as this got merged into the latest nightly during our outbound sync today (4/29/24) - see https://github.com/modularml/mojo/commit/baca7c0928ba5879e6a4c3e4dfa0307d563c8bd5.

**Pull request closed without merging**